lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/launch/REEFMessageCodec.java

```diff
@@ -52,9 +52,7 @@ private REEFMessageCodec() {
   public GeneratedMessage decode(final byte[] bytes) {
     try {
       final REEFProtocol.REEFMessage message = REEFProtocol.REEFMessage.parseFrom(bytes);
-      if (message.hasJobSubmission()) {
-        return message.getJobSubmission();
-      } else if (message.hasJobControl()) {
+      if (message.hasJobControl()) {
```

Comment:
There were no instances of `JobSubmission` actually being attached to a `REEFMessage` so it should be safe to remove it. @bgchun @markusweimer, any arguments against?